File: src/main/java/org/radarcns/consumer/realtime/condition/LocalJsonPathCondition.java
package org.radarcns.consumer.realtime.condition;

import java.io.IOException;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.radarcns.config.realtime.ConditionConfig;

/**
 * Reads the JsonPath specification from the configuration file, provided to this class in {@link
 * ConditionConfig}.
 */
public class LocalJsonPathCondition extends JsonPathCondition {

  public static final String NAME = "LocalJsonPathCondition";

  private final String jsonPath;

  public LocalJsonPathCondition(ConditionConfig conditionConfig) {
    this.jsonPath = (String) conditionConfig.getProperties().get("jsonpath");
    if (jsonPath == null) {
      throw new IllegalArgumentException(
          "The 'jsonpath' property needs to be specified when "
              + "using the LocalJsonPathCondition.");
    }
  }

  @Override
  public String getName() {
    return NAME;
  }

  @Override
  public Boolean isTrueFor(ConsumerRecord<?, ?> record) throws IOException {
    return evaluateJsonPath(record, jsonPath);
  }
}

File: src/main/java/org/radarcns/consumer/realtime/condition/LocalJsonPathCondition.kt
package org.radarcns.consumer.realtime.condition

import org.apache.kafka.clients.consumer.ConsumerRecord
import org.radarcns.config.realtime.ConditionConfig
import java.io.IOException

/**
 * Reads the JsonPath specification from the configuration file, provided to this class in [ ].
 */
class LocalJsonPathCondition(
        conditionConfig: ConditionConfig,
        override val name: String = NAME,
) : JsonPathCondition(conditionConfig) {
    private val jsonPath: String?
    private val rootKey: String?

    @Throws(IOException::class)
    override fun isTrueFor(record: ConsumerRecord<*, *>?): Boolean {
        return evaluateJsonPath(record!!, jsonPath, rootKey)
    }

    companion object {
        const val NAME = "LocalJsonPathCondition"
    }

    init {
        rootKey = conditionConfig.properties?.let { it.getOrDefault("key", null) as String? }
        jsonPath = requireNotNull(
                conditionConfig.properties?.let { it["jsonpath"] as String? }
        ) {
            ("The 'jsonpath' property needs to be specified when "
                    + "using the LocalJsonPathCondition.")
        }
    }
}